source/common/buffer/buffer_impl.cc

```diff
@@ -224,7 +224,7 @@ RawSlice OwnedImpl::frontSlice() const {
 }
 
 SliceDataPtr OwnedImpl::extractMutableFrontSlice() {
-  RELEASE_ASSERT(length_ > 0, "Extract called on empty buffer");
```

Comment:
Doesn't this changes predictable abort to an undefined behavior when `slices_.front() ` is called below?

Reply:
reverted this one - it's not even used in house.  I wonder if we can revert the code and ask the implementor to use a friend helper class...